Directory.mk

```diff
@@ -35,36 +35,30 @@
 
 include $(APPDIR)/Make.defs
 
-# Sub-directories
+# Sub-directories that have been built.
 
-SUBDIRS = $(dir $(wildcard */Makefile))
+CLEANSUBDIRS     = $(dir $(wildcard */.built))
+DISTCLEANSUBDIRS = $(dir $(wildcard */.depend))
 
 all: nothing
 
-.PHONY: nothing context depend clean distclean
+.PHONY: nothing clean distclean
 
-$(foreach SDIR, $(SUBDIRS), $(eval $(call SDIR_template,$(SDIR),preconfig)))
-$(foreach SDIR, $(SUBDIRS), $(eval $(call SDIR_template,$(SDIR),context)))
-$(foreach SDIR, $(SUBDIRS), $(eval $(call SDIR_template,$(SDIR),depend)))
-$(foreach SDIR, $(SUBDIRS), $(eval $(call SDIR_template,$(SDIR),clean)))
-$(foreach SDIR, $(SUBDIRS), $(eval $(call SDIR_template,$(SDIR),distclean)))
+$(foreach SDIR, $(CLEANSUBDIRS), $(eval $(call SDIR_template,$(SDIR),clean)))
+$(foreach SDIR, $(DISTCLEANSUBDIRS), $(eval $(call SDIR_template,$(SDIR),distclean)))
```

Comment:
Should these be moved to a make target so they can use parallelism of '-j'?

Reply:
@btashton the current form already run parallelly.

Reply:
@xiaoxiang781216  I was under the impression that a foreach with eval call would not be run with j parallelism

Reply:
foreach SDIR_template with will expand to something like this:
```
xxx_clean:
  make -C xxx clean
```
And the follow line:
```
clean: $(foreach SDIR, $(CLEANSUBDIRS), $(SDIR)_clean)
```
will expand to:
```
clean: xxx_clean ...
```
so make will execute xxx_clean concurrently.